crates/sargon/src/system/sargon_os/pre_authorization/sargon_os_create_subintent.rs

```diff
@@ -0,0 +1,250 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Create Subintent
+// ==================
+impl SargonOS {
+    /// Creates a Subintent given its discriminator, manifest and expiration.
+    pub async fn create_subintent(
+        &self,
+        intent_discriminator: IntentDiscriminator,
+        subintent_manifest: SubintentManifest,
+        expiration: DappToWalletInteractionSubintentExpiration,
+    ) -> Result<Subintent> {
+        let network_id = self.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+        let current_epoch = gateway_client.current_epoch().await?;
+
+        // Calculate the seconds until the expiration of the subintent.
+        let expiry_time_from_now_in_seconds =
+            self.expiry_time_from_now_in_seconds(expiration);
+        if expiry_time_from_now_in_seconds == 0 {
+            return Err(CommonError::SubintentExpired);
+        }
+
+        // Calculate header ranges
+        let end_ranges = self.calculate_end_ranges(
+            current_epoch,
+            expiry_time_from_now_in_seconds,
+        );
+
+        // Build header
+        let header = IntentHeaderV2 {
+            network_id,
+            start_epoch_inclusive: current_epoch,
+            end_epoch_exclusive: end_ranges.0,
+            min_proposer_timestamp_inclusive: None,
+            max_proposer_timestamp_exclusive: Some(end_ranges.1),
+            intent_discriminator,
+        };
+
+        // Build subintent
+        Subintent::new(header, subintent_manifest, MessageV2::None)
```

Comment:
Message should not be none, it should be set to whatever message was received with the WalletInteraction